### ISSUE TITLE
Anticheat (question)

### ISSUE THREAD

**Issue by mathewcook633** [ufrisk/pcileech-fpga#8]
I saw you commented that this could be used to bypass anti cheats. However, is this viable, considering I can’t imagine finding the values within memory would be particularly quick. Say at physical memory 0x0040377426 is the value of if local player is in the air. Sure reading this value: dump 0x0040377426 0x0040377426 + sizeof(bool) or however the command is done. However, getting the physical address in the first place seems to be where I imagine it would be slow. Unless you have a driver on the host to convert virtual to physical. 

How would you suggest to do this? Also for a Windows 10 target and I’d ideally use a Linux attacker. 

Thunderbolt to pcie express -> PCIe express to PCIe -> fpga board -> raspberry pi or another smaller computer. Or even change up the code for the fpga board so no attacked is required and the processing is done via the fpga.

Would the pciescreamer be stable enough for this or not really. Thanks!

Thanks!

**Comment by ufrisk** [comment]
No, you are correct, the current latency (best case) for small reads are in the millisecond range. (not considering load/setup times for PCILeech the initial USB connection).

Please look into my memory process file system - it mounts virtual memory (translates virtual2physical on the fly using page table walks). As-is right now I suspect it's quite useless though since I added so much caching to it that it will most probably be useless for this scenario. I suspect that if the caching for virtual memory read/write were to be removed (not page table caching) it will be useful, even though I suspect it would still be in the 10ms range. http://blog.frizk.net/2018/03/memory-process-file-system.html

If you need to move beyond ms the way to go about it is probably to load some code onto the FPGA board itself. It's an obvious future development but I suspect there will be so much work around it that I won't be able to do it due to time constraints. This would move latency down to 100uS range.

The PCIeScreamer may be stable enough if you are lucky with your system; I suspect you will run into problems for this use scenario though.
